apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -230,20 +231,28 @@ Task<long> ISpannerTransaction.ExecuteDmlAsync(ExecuteSqlRequest request, Cancel
             request.Seqno = Interlocked.Increment(ref _lastDmlSequenceNumber);
             return ExecuteHelper.WithErrorTranslationAndProfiling(async () =>
             {
-                ResultSet resultSet = await _session.ExecuteSqlAsync(request, timeoutSeconds, cancellationToken).ConfigureAwait(false);
-                var stats = resultSet.Stats;
-                if (stats == null)
+                using (var reader = _session.ExecuteSqlStreamReader(request, timeoutSeconds))
```

Comment:
Maybe add a comment saying, "as requested by the Spanner team", for me at least, finding a stream reader being used for executing DML would be really weird.

Reply:
Done (with a bit more context in fact). Will merge on green.